packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -476,7 +482,23 @@ const DataGrid = injectIntl((props) => {
     if (!event.currentTarget.contains(event.relatedTarget)) {
       // Not triggered when swapping focus between children
       if (handleFocus.current) {
-        setFocusedRowCol(focusedRow, focusedCol, true);
+        // When modifying the number of rows, we set the focused cell to the last selected RowId and columnId
+        if (lastActiveRowId.current && lastActiveColumnId.current) {
```

Comment:
I would think that you would deal with the row and column separately.  I am not sure we need both to exist to operate on one.